### PR TITLE
Small documentation fixes

### PR DESCRIPTION
**Pull request by infraredgirl**
- Add `pick-instance-type` to utility binaries table
- Mention Go style guide in contribution docs
- Fix broken links
- Fix typos

### CODE REVIEW

File: docs/_docs/02_testing-best-practices/iterating-locally-using-docker.md [hunk 1]
```diff
@@ -19,13 +19,13 @@ Here are some techniques we use with Docker:
 
 - If your script is used in a Packer template, add a [Docker
   builder](https://www.packer.io/docs/builders/docker.html) to the template so you can create a Docker image from the
-  same code. See the [Packer Docker Example]({{ site.baseurl }}/examples/code-examples/packer-docker-example/) for working sample code.
+  same code. See the [Packer Docker Example](https://github.com/gruntwork-io/terratest/tree/master/examples/packer-docker-example) for working sample code.
```
Comment (brikis98) [inline docs/_docs/02_testing-best-practices/iterating-locally-using-docker.md:22]: Huh... I could've sworn we had a way to direct-link to the [examples on this page](https://terratest.gruntwork.io/examples/), but I guess not!

**Review by brikis98 — DISMISSED** [review]
LGTM, thanks!

File: docs/_docs/02_testing-best-practices/iterating-locally-using-docker.md (hunk 1 above)
Comment (infraredgirl) [reply docs/_docs/02_testing-best-practices/iterating-locally-using-docker.md:22]: I looked and looked and couldn't find it. Then I just resorted to linking to GH. Better than a broken link. :smile:

Comment (brikis98) [reply docs/_docs/02_testing-best-practices/iterating-locally-using-docker.md:22]: Agreed!

**Review by brikis98 — APPROVED** [review]
LGTM!

**Comment by infraredgirl** [comment]
The test failures are obviously unrelated to this purely doc change, so I'll go ahead and merge (even though I hate merging when the CI is red).

**Pull request merged**